src/sentry/options/manager.py

```diff
@@ -221,6 +223,11 @@ def register(self, key, default=None, type=None, flags=DEFAULT_FLAGS,
             default = type
             default_value = default()
 
+        # Boolean values need to be set to ALLOW_EMPTY becaues otherwise, "False"
+        # would be treated as a not valid value
+        if default_value in (True, False):
```

Comment:
you should do `default_value is True or default_value is False` as `1 in (True, False)` is True.